### PR TITLE
Removing star exports from @fluentui/react-badge

### PR DESCRIPTION
**Pull request by khmakoto**
## Current Behavior

`react-badge` has `export * from ...` in `src/index.ts`.

## New Behavior

`react-badge` has explicitly named exports in `src/index.ts`.

## Related Issue(s)

#22099

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/80746a7f3462c2fa4ff15a92d6e196c67d652f6a'>80746a7f3462c2fa4ff15a92d6e196c67d652f6a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by ecraig12345** [comment]
Build is failing due to a bunch of the exports being duplicated (which I also see looking at the file)

**Review by ling1726 — APPROVED** [review]

File: packages/react-badge/src/index.ts [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+export {
+  Badge,
+  badgeClassName,
+  renderBadge_unstable,
+  useBadgeStyles_unstable,
+  useBadge_unstable,
+} from './components/Badge/index';
+export type { BadgeProps, BadgeSlots, BadgeState } from './components/Badge/index';
+export { Badge, badgeClassName, renderBadge_unstable, useBadgeStyles_unstable, useBadge_unstable } from './Badge';
+export type { BadgeProps, BadgeSlots, BadgeState } from './Badge';
```
Comment (ling1726) [inline packages/react-badge/src/index.ts:10]:
```suggestion
```

File: packages/react-badge/src/index.ts [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+  useCounterBadge_unstable,
+} from './components/CounterBadge/index';
+export type { CounterBadgeProps, CounterBadgeState } from './components/CounterBadge/index';
+export {
+  CounterBadge,
+  counterBadgeClassName,
+  useCounterBadgeStyles_unstable,
+  useCounterBadge_unstable,
+} from './CounterBadge';
+export type { CounterBadgeProps, CounterBadgeState } from './CounterBadge';
```
Comment (ling1726) [inline packages/react-badge/src/index.ts:31]:
```suggestion
```

File: packages/react-badge/src/index.ts [hunk 3]
```diff
@@ -1,5 +1,31 @@
...
+export type { BadgeProps, BadgeSlots, BadgeState } from './components/Badge/index';
+export { Badge, badgeClassName, renderBadge_unstable, useBadgeStyles_unstable, useBadge_unstable } from './Badge';
+export type { BadgeProps, BadgeSlots, BadgeState } from './Badge';
+export {
+  PresenceBadge,
+  presenceBadgeClassName,
+  usePresenceBadgeStyles_unstable,
+  usePresenceBadge_unstable,
+} from './PresenceBadge';
+export type { PresenceBadgeProps, PresenceBadgeState, PresenceBadgeStatus } from './PresenceBadge';
```
Comment (ling1726) [inline packages/react-badge/src/index.ts:17]:
```suggestion
export {
  PresenceBadge,
  presenceBadgeClassName,
  usePresenceBadgeStyles_unstable,
  usePresenceBadge_unstable,
} from './PresenceBadge/index';
export type { PresenceBadgeProps, PresenceBadgeState, PresenceBadgeStatus } from './PresenceBadge/index';
```

We only allow exports from files I believe

**Comment by khmakoto** [comment]
I don't know how this became so messed up 😆

**Review by ecraig12345 — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/Avatar-mount_base.html">1070</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/Avatar-mount.html">1048</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/Button-mount_base.html">644</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/Button-mount.html">616</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProvider-mount_base.html">2100</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProvider-mount.html">2083</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProviderWithTheme-mount_base.html">303</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProviderWithTheme-mount.html">310</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">254</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">272</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">348</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">336</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/MakeStyles-mount_base.html">1853</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22114/perf-test-react-components/MakeStyles-mount.html">1867</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-badge/src/index.ts [hunk 4]
```diff
@@ -1,5 +1,31 @@
-export * from './components/Badge/index';
-export * from './Badge';
-export * from './PresenceBadge';
-export * from './components/CounterBadge/index';
-export * from './CounterBadge';
+export {
+  Badge,
+  badgeClassName,
```
[the remark this answers is not in the document]
Comment (khmakoto) [reply packages/react-badge/src/index.ts:4]: Added.

**Pull request merged**